File: Data Structures & Algorithms/daily-temperatures/submission-0.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        // next greateer element
        stack<int>st;
        int n = temperatures.size();
        vector<int>ans(n,0);
        for(int i=0;i<n;i++){
            while(!st.empty() and temperatures[st.top()]<temperatures[i]){
                ans[st.top()]=(i-st.top());
                st.pop();
            }
            st.push(i);
        }
        return ans;
    }
};

File: Data Structures & Algorithms/subsets/submission-0.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        ans=[]
        fina_ans=[]
        def ps(nums,ans,i):
            if i==len(nums):
                fina_ans.append(ans.copy())
                return
            ans.append(nums[i])
            ps(nums,ans,i+1)
            ans.pop()
            ps(nums,ans,i+1)
        ps(nums,ans,0)
        return fina_ans

                          
        